tests/utils.py

```diff
@@ -87,3 +87,21 @@ def check_server_ready(protocol="grpc"):
         output = buf.getvalue()
     output = json.loads(output)
     return output["ready"]
+
+
+# Context Manager to start and kill a mock server running in background and used by testing functions
+class MockServer:
+    def __init__(self, repo=None, mode="local", timeout=60):
+        self.repo = repo
+        self.mode = mode
+        self.timeout = timeout
+
+    def __enter__(self):
+        run(["remove", "-m", "all"])
+        self.pid = run_server(self.repo, self.mode)
+        wait_for_server_ready()  # Polling
```

Comment:
I think you need to pass `self.timeout` here

Reply:
Thanks for the catch!